pkg/gencred/cmd/main.go

```diff
@@ -0,0 +1,210 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	flag "github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api/latest"
+	"k8s.io/test-infra/pkg/gencred"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	// defaultContextName is the default context name.
+	defaultContextName = "build"
+	// defaultConfigFileName is the default kubeconfig filename.
+	defaultConfigFileName = "kubeconfig.yaml"
+)
+
+// Command-line flags
+var context string
+var name string
+var output string
+var certificate bool
+var serviceaccount bool
+var overwrite bool
+
+// validateFlags validates the command-line flags.
+func validateFlags() {
+	var err error
+
+	if len(name) == 0 {
+		printErrAndExit("-n, --name option is required.", 1)
+	}
+
+	output, err = filepath.Abs(output)
+	if err != nil {
+		printErrAndExit(fmt.Sprintf("-o, --output option invalid: %v.", output), 1)
+	}
+
+	if dirExists(output) {
+		printErrAndExit(fmt.Sprintf("-o, --output already exists and is a directory: %v.", output), 1)
+	}
+
+	if serviceaccount && certificate {
+		printErrAndExit("-c, --certificate and -s, --serviceaccount are mutually exclusive options.", 1)
+	}
+}
+
+// printErr prints error to stderr.
+func printErr(msg string) {
+	_, _ = fmt.Fprintln(os.Stderr, msg)
+}
+
+// printErrAndExit prints error to stderr and exits with te provided code.
+func printErrAndExit(msg string, code int) {
+	printErr(msg)
+	os.Exit(code)
+}
+
+// dirExists checks if a path exists and is a directory.
+func dirExists(path string) bool {
+	info, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return info.IsDir()
+}
+
+// fileExists checks if a path exists and is a file.
+func fileExists(path string) bool {
+	info, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return !info.IsDir()
+}
+
+// mergeConfigs merges an existing kubeconfig file with a new entry with precedence given to the existing config.
+func mergeConfigs(kubeconfig *[]byte) {
```

Comment:
I would expect this function's signature to be more like: `func appendConfig(kc []byte) ([]byte, error)`

AKA: `foo = append(foo, 1)`

Try to avoid pointers to slices when possible